Cython/Utility/ModuleSetupCode.c

```diff
@@ -860,7 +860,7 @@ static CYTHON_INLINE void *__Pyx_PyModule_GetState(PyObject *op)
 #define __Pyx_PyObject_GetSlot(obj, name, func_ctype)  __Pyx_PyType_GetSlot(Py_TYPE((PyObject *) obj), name, func_ctype)
 #define __Pyx_PyObject_TryGetSlot(obj, name, func_ctype) __Pyx_PyType_TryGetSlot(Py_TYPE(obj), name, func_ctype)
 #define __Pyx_PyObject_TryGetSubSlot(obj, sub, name, func_ctype) __Pyx_PyType_TryGetSubSlot(Py_TYPE(obj), sub, name, func_ctype)
-#if CYTHON_COMPILING_IN_LIMITED_API // || !CYTHON_USE_TYPE_SLOTS
+#if CYTHON_COMPILING_IN_LIMITED_API
```

Comment:
Just looking at this - I think this is pretty much the definition of the choice for `!CYTHON_USE_TYPE_SLOTS`. To me that suggests that the right change is that PyPy shouldn't set `CYTHON_USE_TYPE_SLOTS 0`.

A look at the rest of the code seems to support this - there's a lot of cases of `#if CYTHON_USE_TYPE_SLOTS || CYTHON_COMPILING_IN_PYPY` which really just suggests that PyPy does actually want the direct slot access. 

I'll try to make that change as a small separate PR though since it'll probably be clearer that way.